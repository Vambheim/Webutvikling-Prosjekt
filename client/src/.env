# Måte å tilby kryptert API-nøkkel (gir tryggere kommunikasjon mot nettet) 
REACT_APP_API_KEY=5b7869e75f75481a8f5005bc7bffd3a8 
