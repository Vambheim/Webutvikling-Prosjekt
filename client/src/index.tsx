import ReactDOM from 'react-dom';
import * as React from 'react';
import { Component } from 'react-simplified';
import { HashRouter, Route } from 'react-router-dom';
import { Alert, Card } from './widgets';
import {
  RecipeAdd,
  RecipeList,
  ShoppingList,
  TaskDetails,
  TaskEdit,
  TaskNew,
} from './recipe-components';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import NavDropdown from 'react-bootstrap/NavDropdown';
// import Card from 'react-bootstrap/Card';

class Menu extends Component {
  render() {
    return (
      <Navbar bg="light" expand="lg">
        <Container>
          <Navbar.Brand href="/" className="ms-auto">
            Java matapplikasjon
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="ms-auto">
              <Nav.Link href="/#/recipes">Oppskrifter</Nav.Link>
              <Nav.Link href="/#/recipes/add">Legg til oppskrift</Nav.Link>
              <Nav.Link href="/#/recipes/cart">Handleliste</Nav.Link>
              <NavDropdown title="Innstillinger" id="basic-nav-dropdown">
                <NavDropdown.Item href="/#/recipes/cart">Profil</NavDropdown.Item>
                <NavDropdown.Item href="#action/3.2">MÃ¸rk modus</NavDropdown.Item>
                <NavDropdown.Divider />
                <NavDropdown.Item href="#action/3.4">Overrask meg</NavDropdown.Item>
              </NavDropdown>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    );
  }
}

class Home extends Component {
  render() {
    return <Card title="Welcome">This is your favourite food recipe app</Card>;
  }
}

ReactDOM.render(
  <HashRouter>
    <div>
      <Alert />
      <Menu />
      <Route exact path="/" component={Home} />
      <Route exact path="/recipes" component={RecipeList} />
      <Route exact path="/recipes/add" component={RecipeAdd} />
      <Route exact path="/recipes/cart" component={ShoppingList} />
      <Route exact path="/tasks/:id(\d+)" component={TaskDetails} /> {/* id must be number */}
      <Route exact path="/tasks/:id(\d+)/edit" component={TaskEdit} /> {/* id must be number */}
      <Route exact path="/tasks/new" component={TaskNew} />
    </div>
  </HashRouter>,
  document.getElementById('root')
);
