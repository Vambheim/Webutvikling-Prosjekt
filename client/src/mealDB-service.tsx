import axios from 'axios';

axios.defaults.baseURL = 'https://www.themealdb.com/api/json/v1/1';

export type MealDBRecipe = {
  mealDBId: number;
  name: string;
  category: string;
  country: string;
};

class MealDBService {
  /**
   * Get task with given id.
   */
  get(mealDBId: number) {
    return axios.get<MealDBRecipe>('/lookup.php?i=' + mealDBId).then((response) => response.data);
  }

  /**
   * Get all tasks.
   */
  // getAll() {
  //   return axios.get<Recipe[]>('/recipes').then((response) => response.data);
  // }

  /**
   * Create new task having the given title.
   *
   * Resolves the newly created task id.
   */
  // create(title: string) {
  //   return axios
  //     .post<{ id: number }>('/tasks', { title: title })
  //     .then((response) => response.data.id);
  // }
}

const mealDBService = new MealDBService();
export default mealDBService;
