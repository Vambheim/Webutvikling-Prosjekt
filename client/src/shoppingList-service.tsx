import axios from 'axios';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

export type ShoppingListInfo = {
  shopping_list_id: number;
  recipe_id: number;
  ingredient_id: number;
  name: string;
  amount: number;
  measurement_unit: string;
};

class ShoppingListService {
  /**
   * Get shopping list for given user
   */
  getShoppingList(user_id: number) {
    return axios
      .get<ShoppingListInfo[]>('/shoppinglist/' + user_id)
      .then((response) => response.data);
  }

  /**
   * Add to shopping list for given user
   */
  addToShoppingList(
    recipe_id: number,
    ingredient_id: number,
    user_id: number,
    amount: number,
    measurement_unit: string
  ) {
    return axios
      .post('/shoppinglist', {
        recipe_id: recipe_id,
        ingredient_id: ingredient_id,
        user_id: user_id,
        amount: amount,
        measurement_unit: measurement_unit,
      })
      .then((response) => response.data);
  }

  /**
   * Delete whole shopping list for given user
   */
  deleteShoppingList(user_id: number) {
    return axios.delete('/shoppinglist/' + user_id).then((response) => response.data);
  }

  /**
   * Delete one item in shopping list
   */
  deleteItemShoppingList(shopping_list_id: number) {
    return axios.delete('/shoppinglistitem/' + shopping_list_id).then((response) => response.data);
  }
}

const shoppingListService = new ShoppingListService();
export default shoppingListService;
