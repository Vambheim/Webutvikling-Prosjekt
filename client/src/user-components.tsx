import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Column } from './widgets';
import { Button, Form, Card, Row, Col, Container } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';
import recipeService, { Recipe } from './recipe-service';
import userService, { User } from './user-service';
import { createHashHistory } from 'history';

//false as default
export let loggedIn: boolean = false;
export let currentUser: User = {
  user_id: 0,
  email: '',
  first_name: '',
  last_name: '',
  password: '',
};

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a recipe

export class UserLogIn extends Component {
  email: string = '';
  password: string = '';

  render() {
    return (
      <Card
        style={{
          border: 'none',
          padding: '15px',
          textAlign: 'center',
          marginLeft: 'auto',
          marginRight: 'auto',
        }}
      >
        {/*Card forms in for log in screen */}
        <Card.Title>Log in</Card.Title>
        <Container style={{ width: '20rem', marginLeft: 'auto', marginRight: 'auto' }}>
          <Row>
            <Form.Control
              value={this.email}
              type="text"
              placeholder="Email"
              onChange={(event) => (this.email = event.currentTarget.value)}
              style={{
                textAlign: 'center',
                marginBottom: '10px',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.password}
              type="password"
              placeholder="Password"
              onChange={(event) => (this.password = event.currentTarget.value)}
              // Makes it possible to log in with enter as well as with button
              onKeyUp={(event) => {
                if (event.key == 'Enter') {
                  this.logIn();
                }
              }}
              style={{
                textAlign: 'center',
                marginBottom: '10px',
              }}
            ></Form.Control>
          </Row>
        </Container>
        {/*Card for buttons in login screen before user is identified or registered */}
        <Container style={{ width: '15rem', marginLeft: 'auto', marginRight: 'auto' }}>
          <Row>
            <Button
              variant="success"
              onClick={() => this.logIn()}
              style={{
                marginBottom: '10px',
              }}
            >
              Log in
            </Button>
          </Row>
          <Row>
            <Button
              variant="outline-success"
              onClick={() => this.createUser()}
              style={{
                marginBottom: '10px',
              }}
            >
              No user? Create one here
            </Button>
          </Row>
          <Row>
            <Button
              variant="outline-success"
              onClick={() => this.clearInput()}
              style={{
                marginBottom: '10px',
              }}
            >
              Clear input
            </Button>
          </Row>
        </Container>
      </Card>
    );
  }

  logIn() {
    if (this.email.length != 0 && this.password.length != 0) {
      userService
        .logIn(this.email, this.password)
        .then((user) => {
          currentUser = user;
          loggedIn = true;
          Alert.success('Logged in as ' + currentUser.email);
          history.push('/recipes/user');
        })
        .catch((error) => Alert.danger(error.response.data));
    } else {
      Alert.danger('Please fill in all the fields');
    }
  }

  clearInput() {
    this.email = '';
    this.password = '';
  }

  createUser() {
    history.push('/recipes/register');
  }
}

export class RegisterUser extends Component {
  user: User = { user_id: 0, email: '', first_name: '', last_name: '', password: '' };
  confirm_password: string = '';

  render() {
    return (
      <Card
        style={{
          border: 'none',
          padding: '15px',
          textAlign: 'center',
          marginLeft: 'auto',
          marginRight: 'auto',
        }}
      >
        {/* Card creating forms related to creating new user */}
        <Card.Title>Create user</Card.Title>
        <Container
          style={{
            marginLeft: 'auto',
            marginRight: 'auto',
            width: '20rem',
          }}
        >
          <Row>
            <Form.Control
              value={this.user.email}
              type="text"
              placeholder="Email"
              onChange={(event) => (this.user.email = event.currentTarget.value)}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.user.first_name}
              type="text"
              placeholder="First name"
              onChange={(event) => (this.user.first_name = event.currentTarget.value)}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.user.last_name}
              type="text"
              placeholder="Last name"
              onChange={(event) => (this.user.last_name = event.currentTarget.value)}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.user.password}
              type="password"
              placeholder="Password"
              onChange={(event) => (this.user.password = event.currentTarget.value)}
              // Makes it possible to log in with enter as well as with button
              onKeyUp={(event) => {
                if (event.key == 'Enter') {
                  this.createUser();
                }
              }}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
          <Row>
            <Form.Control
              value={this.confirm_password}
              type="password"
              placeholder="Confirm password"
              onChange={(event) => (this.confirm_password = event.currentTarget.value)}
              onKeyUp={(event) => {
                if (event.key == 'Enter') {
                  this.createUser();
                }
              }}
              style={{
                marginBottom: '10px',
                textAlign: 'center',
              }}
            ></Form.Control>
          </Row>
        </Container>
        {/* Buttons for creating user and clearing input */}
        <Container style={{ width: '15rem', marginLeft: 'auto', marginRight: 'auto' }}>
          <Row>
            <Button
              variant="success"
              onClick={() => this.createUser()}
              style={{
                marginBottom: '10px',
              }}
            >
              Create user
            </Button>
          </Row>
          <Row>
            <Button
              variant="outline-success"
              onClick={() => this.clearInput()}
              style={{
                marginBottom: '10px',
              }}
            >
              Clear input
            </Button>
          </Row>
        </Container>
      </Card>
    );
  }

  createUser() {
    userService
      .createUser(
        this.user.email,
        this.user.first_name,
        this.user.last_name,
        this.user.password,
        this.confirm_password
      )
      .then((response) => {
        if (response.length > 0) {
          Alert.danger(response);
        } else {
          Alert.success('User created, please log in');
          loggedIn = true;
          history.push('/recipes/login');
        }
      })
      .catch((error) => Alert.danger(error.response.data));
  }

  clearInput() {
    this.user = { user_id: 0, email: '', first_name: '', last_name: '', password: '' };
    this.confirm_password = '';
  }
}

export class UserDetails extends Component {
  likedRecipes: Recipe[] = [];
  render() {
    return (
      <>
        <Card
          style={{
            // border: 'none',
            padding: '15px',
            textAlign: 'center',
            marginLeft: 'auto',
            marginRight: 'auto',
          }}
        >
          {/* Page for all relevant user info for logged in user */}
          <Card.Title>
            {'User page for ' + currentUser.first_name + ' ' + currentUser.last_name}
          </Card.Title>
          <Row style={{ fontSize: '17px' }}>
            <Card.Text>Your email-adress: {currentUser.email}</Card.Text>
          </Row>
          <Row style={{ fontSize: '17px' }}>
            <Card.Text>
              Your name: {currentUser.first_name} {currentUser.last_name}
            </Card.Text>
          </Row>
          <Row>
            <Button
              variant="outline-danger"
              onClick={() => this.logOut()}
              style={{
                width: '15rem',
                marginLeft: 'auto',
                marginRight: 'auto',
                marginBottom: '10px',
              }}
            >
              Log out
            </Button>
          </Row>
        </Card>

        {/* Card retrieving active users liked recipes */}
        <Card
          style={{
            border: 'none',
            padding: '15px',
            textAlign: 'center',
            marginLeft: 'auto',
            marginRight: 'auto',
          }}
        >
          <Card.Title>Your liked recipes</Card.Title>
          <Row>
            <Col lg>
              <Row xs={1} md={4} className="g-4">
                {this.likedRecipes.map((recipe) => (
                  <NavLink
                    to={'/recipes/' + recipe.recipe_id}
                    style={{
                      color: '#9FC1C0',
                    }}
                  >
                    <Column>
                      <Card
                        style={{
                          width: '100%',
                          margin: '1%',

                          textAlign: 'center',
                          borderLeft: 'none',
                          borderRight: 'none',
                          borderTop: 'none',
                          borderRadius: 'none',
                        }}
                      >
                        <Card.Img variant="top" src="https://s.tihlde.org/recipechef12312" />
                        <Card.Body>
                          <Card.Title style={{ color: 'rgb(82, 130, 101)' }}>
                            {recipe.name}
                          </Card.Title>
                          <Card.Text>
                            {recipe.country} {recipe.category}
                          </Card.Text>
                        </Card.Body>
                      </Card>
                    </Column>
                  </NavLink>
                ))}
              </Row>
            </Col>
          </Row>
        </Card>
      </>
    );
  }

  mounted() {
    if (!loggedIn) {
      history.push('/recipes/login');
    } else {
      recipeService
        .getLikedRecipes(currentUser.user_id)
        .then((recipes) => (this.likedRecipes = recipes))
        .catch((error) => Alert.danger(error.message));
    }
  }

  logOut() {
    loggedIn = false;
    history.push('/recipes');
    currentUser = { user_id: 0, email: '', first_name: '', last_name: '', password: '' };
  }
}
