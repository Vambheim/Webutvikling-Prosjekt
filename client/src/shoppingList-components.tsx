import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Column } from './widgets';
import shoppingListService, { ShoppingListInfo } from './shoppingList-service';
import { loggedIn, currentUser } from './user-components';
import { createHashHistory } from 'history';
import { Container, Card, Row, Form, Button, Col } from 'react-bootstrap';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a recipe

export class ShoppingList extends Component {
  shopping_list: ShoppingListInfo[] = [];

  render() {
    return (
      // <Container style={{ backgroundColor: 'yellow' }}>
      //   <Card style={{ backgroundColor: 'green' }}>
      //     <Card.Title>Shopping List</Card.Title>

      //     {this.shopping_list.map((list) => (
      //       <Row key={list.shopping_list_id}>
      //         <Column width={3}>
      //           {list.amount + ' ' + list.measurement_unit + ' ' + list.name}
      //         </Column>
      //         <Column width={1}>
      //           <Button
      //             variant="success"
      //             onClick={() => this.deleteOne(list.shopping_list_id, list.name)}
      //           >
      //             &#128465;
      //           </Button>
      //         </Column>
      //       </Row>
      //     ))}

      //     <Button variant="danger" onClick={() => this.deleteAll()}>
      //       Delete
      //     </Button>
      //   </Card>
      // </Container>

      <Card
        style={{
          // border: '5px',
          // borderStyle: ' solid',
          padding: '15px',
          textAlign: 'center',
          // marginLeft: 'auto',
          // marginRight: 'auto',
        }}
      >
        <Card.Title>Shopping list</Card.Title>

        <Row
          style={{
            marginLeft: '7%',
            width: '90%',
          }}
        >
          <Col>
            {this.shopping_list.map((list) => (
              <Row key={list.shopping_list_id}>
                <Col
                  style={{
                    textAlign: 'right',
                  }}
                >
                  {list.amount + ' ' + list.measurement_unit + ' ' + list.name}
                </Col>
                <Col>
                  <Button
                    variant="outline-success"
                    onClick={() => this.deleteOne(list.shopping_list_id, list.name)}
                    style={{
                      width: '5rem',
                      marginLeft: '0px',
                      marginRight: '100%',
                      marginBottom: '10px',
                    }}
                  >
                    &#128465;
                  </Button>
                </Col>
              </Row>
            ))}
          </Col>
        </Row>

        <Row>
          <Button
            variant="danger"
            onClick={() => this.deleteAll()}
            style={{
              width: '15rem',
              marginLeft: 'auto',
              marginRight: 'auto',
              marginBottom: '10px',
            }}
          >
            Delete all
          </Button>
        </Row>
      </Card>
    );
  }

  mounted() {
    if (!loggedIn) {
      Alert.info('Log in to use this feature');
    } else {
      shoppingListService
        .getShoppingList(currentUser.user_id)
        .then((list) => (this.shopping_list = list))
        .catch((error) => Alert.danger('Error getting shopping list ' + error.message));
    }
  }

  deleteOne(list_id: number, name: string) {
    if (confirm('Do you want to remove ' + name + ' from the shopping list?')) {
      console.log(list_id);
      shoppingListService
        .deleteItemShoppingList(list_id)
        .then(() => Alert.success('Item deleted'))
        .then(() => this.mounted()) // refreshes til items in shopping list
        .catch((error) => Alert.danger('Error deleting item in shopping list: ' + error.message));
    } else {
      console.log('Canceled');
    }
  }

  deleteAll() {
    if (!loggedIn) {
      Alert.info('Please log in');
    } else {
      if (this.shopping_list.length > 0) {
        if (confirm('Do you want to remove all items from shopping list?')) {
          shoppingListService
            .deleteShoppingList(currentUser.user_id)
            .then(() => Alert.info('Shopping list was successfully deleted'))
            .then(() => this.mounted())
            .catch((error) => Alert.danger('Error deleting shopping list ' + error.message));
        } else {
          console.log('Canceled');
        }
      } else {
        Alert.info('No items to delete');
      }
    }
  }
}
