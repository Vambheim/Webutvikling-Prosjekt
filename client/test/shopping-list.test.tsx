import * as React from 'react';
import { ShoppingList } from '../src/shoppingList-components';
import { EnzymeAdapter, shallow } from 'enzyme';
import { Component } from 'react-simplified';
import { Alert, Column } from '../src/widgets';
import { ShoppingListInfo } from '../src/shoppingList-service';
import { loggedIn, currentUser } from '../src/user-components';
import { createHashHistory } from 'history';
import { Container, Card, Row, Form, Button, Col } from 'react-bootstrap';

var deleteButtonClicked = false
var knapp1 = false
jest.mock('../src/shoppingList-service', () => {
  class shoppingListService {
    getShoppingList() {
      return Promise.resolve([
        {
          ingredient_id: 2,
          name: "groennbaer",
          recipe_id: 1,
          shopping_list_id: 2,
          amount: 2,
          measurement_unit: "ml"
        },
        {
          ingredient_id: 3,
          name: "roedbaer",
          recipe_id: 1,
          shopping_list_id: 2,
          amount: 4,
          measurement_unit: "ml"
        },
      ])
    }
    addToShoppingList(
      recipe_id: number,
      ingredient_id: number,
      user_id: number,
      amount: number,
      measurement_unit: string
    ) {
      return Promise.resolve();
    }

    deleteShoppingList(user_id: number) {
      deleteButtonClicked = true
      return Promise.resolve();
    }

    deleteItemShoppingList(shopping_list_id: number) {
      knapp1 = true
      return Promise.resolve();
    }
  }
  return new shoppingListService()
})

// @ts-ignore: do not type check next line.
const wrapper = shallow(<ShoppingList />);

describe('Shoppinglist component tests', () => {
  test('Shoppinglist draws correctly (using snapshot)', (done) => {

    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  })
})

describe('Shoppinglist Button test', () => {
  test('testing if delete button draws', (done) => {

    // @ts-ignore: do not type check next line.
    const button = wrapper.find(Button).at(0)

    // expect(button.length).toBe(1)
    button.simulate('click')

    setTimeout(() => {
      //expect(location.hash).toEqual("#/recipes/shoppinglist")
      expect(deleteButtonClicked || knapp1).toEqual(true)
      done();
    });

    //expect(location.hash).toEqual('#/');


    //   setTimeout(() => {
    //     expect(location.hash).toEqual('');

    //     done();
    //   }, 1000);

  })

  test('testing if first buttons draws', (done) => {
    let buttonsClicked: Array<Boolean> = []
    let getShoppingList = [
      {
        ingredient_id: 2,
        name: "groennbaer",
        recipe_id: 1,
        shopping_list_id: 2,
        amount: 2,
        measurement_unit: "ml"
      },
      {
        ingredient_id: 3,
        name: "roedbaer",
        recipe_id: 1,
        shopping_list_id: 2,
        amount: 4,
        measurement_unit: "ml"
      },
    ]

    // @ts-ignore: do not type check next line.
    const shoppinglist = shallow((
      <Col>
        {getShoppingList.map((list) => (
          <Row key={list.shopping_list_id}>
            <Col
              style={{
                textAlign: 'right',
              }}
            >
              {list.amount + ' ' + list.measurement_unit + ' ' + list.name}
            </Col>
            <Col>
              <Button
                variant="light"
                onClick={() => buttonsClicked.push(true)}
                style={{
                  width: '5rem',
                  marginLeft: '0px',
                  marginRight: '100%',
                  marginBottom: '10px',
                }}
              >
                &#128465;
              </Button>
            </Col>
          </Row>
        ))}
      </Col>
    ));

    // @ts-ignore: do not type check next line.
    const button1 = shoppinglist.find(Button).at(0)

    // @ts-ignore: do not type check next line.
    const button2 = shoppinglist.find(Button).at(1)

    // expect(button.length).toBe(1)
    button1.simulate('click')
    button2.simulate('click')
    //expect(button.length).toBe(1)
    // button.simulate('click')

    setTimeout(() => {
      //expect(location.hash).toEqual("#/recipes/shoppinglist")
      expect(buttonsClicked[0] && buttonsClicked[0]).toEqual(true)
      done();
    });
  })

});
