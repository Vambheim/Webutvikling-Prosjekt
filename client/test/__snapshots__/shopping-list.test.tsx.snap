// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoppinglist component tests Shoppinglist draws correctly (using snapshot) 1`] = `
<Card
  body={false}
  style={
    {
      "border": "none",
      "padding": "15px",
      "textAlign": "center",
    }
  }
>
  <CardTitle>
    Shopping list
  </CardTitle>
  <Row
    style={
      {
        "marginLeft": "7%",
        "width": "90%",
      }
    }
  >
    <Col />
  </Row>
  <Row>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      style={
        {
          "marginBottom": "10px",
          "marginLeft": "auto",
          "marginRight": "auto",
          "width": "15rem",
        }
      }
      variant="danger"
    >
      Delete all
    </Button>
  </Row>
</Card>
`;
